src/fprime_gds/flask/static/js/loader.js

```diff
@@ -49,6 +50,118 @@ export function loadTextFileInputData(event) {
     });
 }
 
+/**
+ * Parser to safely handle potential JSON object from Python. Python can produce some non-standard values (infinities,
+ * NaNs, etc.) These values then break on the JS Javascript parser. To localize these faults, they are replaced before
+ * processing with strings and then formally set during parsing.
+ *
+ * This is done by looking for tokens in unquoted text and replacing them with string representations.
+ *
+ */
+class SaferParser {
+    /**
+     * Set up the parser
+     */
+    constructor() {
+        this.STATES = {
+            UNQUOTED: 0,
+            QUOTED: 1
+        };
+        this.FLAG = "-_-您好"; // Extended character usage make collisions less-likely
+        this.MAPPINGS = [
+            ["-Infinity", this.FLAG + "-inf", -Infinity],
+            ["Infinity", this.FLAG + "inf", Infinity],
+            ["NaN", this.FLAG + "nan", NaN],
+            ["null", this.FLAG + "null", null]
+        ];
```

Comment:
@LeStarch why are we using a flag here? Is it so that if "-inf", "inf", "nan", or "null" appear in a string we could differentiate between that and the substring converted by the SafeParser?

Reply:
Yes.  A better flag-free solution would have required changes on both the Python end and here.  It seemed a localized solution converting to something safe and immediately back was a better solution because:

1. Colocated changes
2. Does not increase size of transmitted data
3. These cases should be sufficiently rare as to not merit a much more complex and comprehensive solution

Reply:
thanks for the explanation! looks good to me!